test-utils/src/main/groovy/ch/tutteli/gradle/plugins/test/Asserts.groovy

```diff
@@ -79,4 +80,14 @@ class Asserts {
         assertEquals(cause, exception.cause.class)
         assertEquals(message, exception.cause.message)
     }
+
+    static void assertTaskExists(Project project, String taskName) {
+        def task = project.tasks.findByName(taskName)
+        if (task == null) {
+            fail("could not find task $taskName")
+            //TODO 5.0.0 check if we can use the following with gradle 8.1.x, instead of the above,
+            // currently we cannot due to the following bug https://github.com/gradle/gradle/issues/20301
+//            fail("could not find task $taskName, there were ${project.tasks.collect { it.name }.join(",")}")
```

Comment:
Have you tried TaskCollection#getNames?

Reply:
thanks for the hint, this one actually works :+1: